docs/current/543069-index.md

```diff
@@ -63,6 +63,69 @@ lib -..-> engine -..-> oci -..-> A1 & B1 & C1
 5. When all operations in the pipeline have been resolved, the engine sends the pipeline result back to your program.
 6. Your program may use the pipeline's result as input to new pipelines.
 
+The engine is made of 2 parts: an API router, and a runner.
+
+* The router serves API queries and dispatches individual operations to the runner.
+* The runner talks to your OCI runtime to execute actual operations. This is basically a buildkit daemon + some glue
+
+The router currently runs on the client machine, whereas the runner is on a worker machine that will run the containers. This could be the same machine but typically isn’t.
+
+```mermaid
+graph LR;
+
+subgraph clientMachineA["Client machine"]
+  clientA["client library or CLI"]
+  routerA["API router"]
+ clientA --> routerA
+end
+subgraph clientMachineB["Client machine"]
+  clientB["client library or CLI"]
+  routerB["API router"]
+ clientB --> routerB
+end
+subgraph clientMachineC["Client machine"]
+  clientC["client library or CLI"]
+  routerC["API router"]
+  clientC --> routerC
+end
+
+
+
+subgraph workerMachine["Worker machine"]
+runner["Runner"]
+oci["OCI runtime"]
+end
+routerA & routerB & routerC -..-> runner --> oci
+```
+
+The SDK takes care of running both router and runner - each in a different way because they run in different machines and have different lifecycles.
+
+Eventually we will move the router to a server-side component, tightly coupled and co-located with the runner. This will be shipped as an OCI image which you will be able to provision, administer and upgrade yourself to your heart’s content. This requires non-trivial engineering work, in order to make the API router multi-tenant.
+
+```mermaid
+graph LR;
+
+subgraph clientMachineA["Client machine"]
+  clientA["client library or CLI"]
+end
+subgraph clientMachineB["Client machine"]
+  clientB["client library or CLI"]
+end
+subgraph clientMachineC["Client machine"]
+  clientC["client library or CLI"]
+end
+
+
+subgraph workerMachine["Worker machine"]
+subgraph engine["Dagger Engine"]
+router["API router"]
+runner["Runner"]
+end
+oci["OCI runtime"]
+end
+clientA & clientB & clientC -..-> router --> runner --> oci
+```
+
```

Comment:
Since this part is only a possible future of Dagger, shall it be mentioned right now in the doc?